### PR TITLE
add:lesson03-add two li, two a and two img in ul.

### PR DESCRIPTION
**Pull request by nakagawamai**
# [Issue3](https://github.com/nakagawamai/js-lesson.git)
## [CodeSandBox](https://codesandbox.io/s/silly-christian-isc7pn?file=/script.js) :[Latest Commit Link](https://github.com/nakagawamai/js-lesson/pull/4/commits/536f863e2eeecc734d5f710c2a88eea7635d6460)
このDOM
````
<ul>
  <li><a href="a1.html"><img src="/img/bookmark.png">a1</a></li>
  <li><a href="a2.html"><img src="/img/message.png">a2</a></li>
</ul>
````
をJavaScriptでつくり、html内のulの中に差し込んでください

## Questions
- I want to know what happens when the number of arrays is different.

### CODE REVIEW

File: lesson03/script.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    const a = document.createElement('a');
+    a.href = aHref[i];
+    a.textContent = aText[i];
+
+    const img = document.createElement('img');
+    img.src = imgSrc[i];
+
+    ul.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+}
```
Comment (nakagawamai) [inline lesson03/script.js]: The following writing method also worked, but which is the better method?
````
for (let i = 0; i < 2 ; i++){
    const li = document.createElement('li');

    const a = document.createElement('a');
    a.href = aHref[i];
    a.textContent = aText[i];

    const img = document.createElement('img');
    img.src = imgSrc[i];

    ul.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
}
````

Comment (csakyo) [inline lesson03/script.js:17]: IMO
Using appendchild in a for loop is not good for performance. 
You can study "createDocumentFragment" on this lesson! This is good study so please refer the Monaka san's article! 

https://monatamalog.com/archives/291

File: lesson03/script.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const aHref = ['a1.html','a2.html'];
+const aText = ['a1','a2'];
+const imgSrc = ['./img/bookmark.png','./img/message.png'];
+
+for (let i= 0; i < aHref.length; i++){
+    const li = document.createElement('li');
+
+    const a = document.createElement('a');
+    a.href = aHref[i];
+    a.textContent = aText[i];
```
Comment (csakyo) [inline lesson03/script.js]: IMO

I think you can wright with "Template literals" here as below. Then the arrays "const aHref" and "const aText" are no longer needed.

Reference:
https://www.javadrive.jp/javascript/string/index5.html

```suggestion
    a.href = `a${i+1}.html`;
    a.textContent = `a${i+1}`;
```


or If you set the initial value here from 1, you won't need "+1".
For example,

``for (let i= 1; i <= liLength; i++){...``
``    a.href = `a${i}.html`;``
 ``   a.textContent = `a${i}`;``

File: lesson03/script.js (hunk 1 above)
Comment (csakyo) [reply lesson03/script.js]: IMO 
``This i < 2`` is a magic number that I don't know what 2 is, so an explanatory variable might be a good idea.

For example, 
``const liLength = 2; ``
``for (let i = 0; i < liLength ; i++){...``

Then it's easy to imagine what number 2 is.

Comment (nakagawamai) [reply lesson03/script.js]: Thank you for your comment!

I got it.
I need to clarify what  number 2 is, isn't it.

Comment (nakagawamai) [reply lesson03/script.js:17]: That is a very important thing.
Thank you for the documentation!
I will study "createDocumentFragment" with Monaka san's article.

File: lesson03/script.js (hunk 2 above)
Comment (nakagawamai) [reply lesson03/script.js]: I don't know "Template literals".
I will study, and challenge to write with  "Template literals".

File: lesson03/script.js [hunk 3]
```diff
@@ -1,17 +1,19 @@
 const ul = document.getElementById('js-list');
-const aHref = ['a1.html','a2.html'];
-const aText = ['a1','a2'];
 const imgSrc = ['./img/bookmark.png','./img/message.png'];
 
-for (let i= 0; i < aHref.length; i++){
+const fragment = document.createDocumentFragment();
```
Comment (nakagawamai) [inline lesson03/script.js:4]: I use "document.createDocumentFragment" to improve performance.

File: lesson03/script.js [hunk 4]
```diff
@@ -1,17 +1,19 @@
...
+
+for (let i= 0; i < imgSrc.length; i++){
     const li = document.createElement('li');
 
     const a = document.createElement('a');
-    a.href = aHref[i];
-    a.textContent = aText[i];
+    a.href = `a${i+1}.html`;
+    a.textContent = `a${i+1}`;
```
Comment (nakagawamai) [inline lesson03/script.js:11]: I use "template literals".
Because to shorten code.

**Review by taka0209 — APPROVED** [review]

**Review by csakyo — APPROVED** [review]

**Pull request merged**